app/dao/organisation_dao.py

```diff
@@ -127,8 +127,12 @@ def dao_get_invited_organisation_user(user_id):
 
 
 def dao_get_users_for_organisation(organisation_id):
-    return User.query.filter(
-        User.organisations.any(id=organisation_id),
+    return db.session.query(
+        User
+    ).join(
+        User.organisations
```

Comment:
don't we have to join on Organisation table here?

Reply:
yeah, it is doing that. User.organisations is a relationship, from `User` to `Organisation`. it's defined as:

```python
    organisations = db.relationship(
        'Organisation',
        secondary='user_to_organisation',
        backref='users')
```

this tells sqlalchemy how to write the sql (going via user_to_organisation), and it identifies the foreign keys and builds up the join statement magically

Reply:
amazing, thanks! :)